src/codegen.rs

```diff
@@ -3278,3 +3302,11 @@ fn postorder_traversal(
 
     output.push(type_id);
 }
+
+fn gen_temp_varaible() -> String {
+    static COUNTER: AtomicUsize = AtomicUsize::new(0);
+    format!(
+        "__temp{}",
+        COUNTER.fetch_add(1, std::sync::atomic::Ordering::SeqCst)
```

Comment:
I'm no expert in rust, but this feels rather unnecessary. There has to be an easier way to just get incrementing numbers, right? We don't really care about being atomic here.

Reply:
Rust doesn't allow mutable global variables without a crate. So it is necessary.